specs/commandline/key.md

```diff
@@ -115,10 +115,22 @@ notation key list
 
 Upon successful execution, a list of keys is printed out with information of name, key path, certificate path, key id and plugin name. The default signing key name is preceded by an asterisk. The key id and plugin name are used together to provide the information of the key identifier for the remote key and the plugin associated with it.
 
-### Delete two keys from signing key list
+### Remove a specified key from signing key list
 
 ```shell
-notation key delete <key_name_1> <key_name_2>
+notation key remove <key_name>
 ```
 
-Upon successful execution, the names of deleted signing keys are printed out. Please be noted if default signing key is deleted, Notation will not automatically assign a new default signing key. User needs to update the default signing key explicitly.
+Upon successful execution, the output message is printed out as following:
+
+```text
+Removed <key_name> from signing key list. You still need to delete the key file from <key_path> and delete the certificate from <cert_path>.
```

Comment:
1. Since we do not support local key, the key created by `generate-test` becomes the only use case of this error message.
2. `You still need to delete the certificate from <cert_path>.` We can show this message only when the key is created by `generate-test`. Because in production, `decide not to Sign with a key anymore` is different from `decide not to trust a certificate in Verification anymore`.

Reply:
Agree 1 and 2. This is against locally creating the test key/cert by `generate-test`.
I will add a prerequisite to explain this scenario in Spec. We also need to differentiate scenarios and explicitly tell users that the remote key/cert can not be deleted by Notation.

Reply:
Yes, and as we discussed, we should introduce `notation generate-test` command to decouple `generate-test` with `cert`. This can be done between RC.4 and v1.0.

Reply:
I agree with Patrick

Reply:
There is a case that if the key to be removed is marked as default key. The output message should be improved as well.

Reply:
@patrickzheng200 I will create another issue to track this new feature request.

Reply:
@yizha1 @patrickzheng200  If a default key is removed from the signing key list, naturally, it means there is no default key anymore. So it's not necessary to tell users the key `xxx` is unmarked as default in the output msg.

Reply:
@FeynmanZhou On a second thought, this actually is not a simple change. When deleting keys, we are deleting them by key name. From key name itself, we cannot tell if that's the key that been generated by `notation cert generate-test` command. In order to do it, we need to also make changes in notation-go library. Since we have agreed to introduce `notation generate-test` command in next release, I suggest we just make a minor change on the error message here in RC.4, something like:
`Removed test key <key_name> from Notation signing key list. The source key file is not deleted.`

Reply:
@patrickzheng200 When using `notation key list`, `key path` is also printed out, maybe we can print out the `The source <key path> is not deleted. `  If it also takes more time, I agree with your proposal. There is also `cert path` in the output message of `notation key list` and seems only useful for local testing keys. We can improve this later with the new `notation generate-test` command.

Reply:
> @patrickzheng200 When using `notation key list`, `key path` is also printed out, maybe we can print out the `The source <key path> is not deleted. ` If it also takes more time, I agree with your proposal. There is also `cert path` in the output message of `notation key list` and seems only useful for local testing keys. We can improve this later with the new `notation generate-test` command.

@yizha1 Good idea, but I think it won't work on Deletion. Here's the reason: `notation key list` reads in the `signingkeys.json` file and print out the keys one by one. When deleting one or more keys, we delete by the key names (the core deletion logic is in notation-go). At the point of printing out, those keys have already been deleted from `signingkeys.json`. In other words, we cannot find the deleted key names anymore by just going through the file again. A workaround would be saving a copy of the whole key list before doing any Deletion. When printing out, we go through that pre-load list. But IMO, this isn't something that we should do just for some print outs. Therefore, I still suggest we use the message: `Removed test key <key_name> from Notation signing key list. The source key file is not deleted.` and introduce `notation generate-test` command in the next release to solve the issue completely.

Reply:
@patrickzheng200 Agree. We could improve it iteratively and aim to fix it by introducing a new command in v1.0.0 completely.

Reply:
Are we renaming the command from `delete` to `remove`? This will be a breaking change - isn't it?